File: packages/java/src/main/java/org/realtimemail/TraditionalMailAccountManager.java
package org.realtimemail;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public final class TraditionalMailAccountManager {
  private final Map<String, TraditionalMailAccount> accounts = new HashMap<>();

  public void addAccount(TraditionalMailAccount account) {
    accounts.put(account.id(), account);
  }

  public void removeAccount(String accountId) {
    accounts.remove(accountId);
  }

  public List<TraditionalMailAccount> listAccounts() {
    return new ArrayList<>(accounts.values());
  }
}

File: packages/csharp/RealtimeMail/TraditionalMailAccountManager.cs
namespace RealtimeMail;

public sealed class TraditionalMailAccountManager
{
    private readonly Dictionary<string, TraditionalMailAccount> accounts = new();

    public void AddAccount(TraditionalMailAccount account)
    {
        accounts[account.Id] = account;
    }

    public void RemoveAccount(string accountId)
    {
        accounts.Remove(accountId);
    }

    public IReadOnlyList<TraditionalMailAccount> ListAccounts()
    {
        return accounts.Values.ToList();
    }
}
